napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -64,27 +98,46 @@ def __init__(
         enabled: bool = True,
         installed: bool = False,
         npe_version=1,
+        versions_conda: list = [],
+        versions_pypi: list = [],
     ):
         super().__init__(parent)
+        self.url = url
+        self._versions = {}
+        self._versions['Conda'] = versions_conda
+        self._versions['PyPI'] = versions_pypi
         self.setup_ui(enabled)
         self.plugin_name.setText(package_name)
+
+        self._populate_version_dropdown('PyPI')
```

Comment:
The same magic constraint here. The `self._populate_version_dropdown` is not annotated with `Literal["PyPI", "Conda"]`